### ISSUE TITLE
Locks up soon after telnet session begins

### ISSUE THREAD

**Issue by reflectiveroot** [FozzTexx/tcpser#4]
I'm using VICE and connecting to tcpser via the ip232 bridge.  It seems that after a telnet session has started, tcpser will hang soon after input is received from the client.

After some debugging, I found select() is failing and invalidating the outputs, but tcpser only logs the error and continues processing.

select() fails because (bridge.c, line 343):

timer.tv_usec = cfg->s[SRegisterGuardTime] * 20000;

needs to be this (or similar):

timer.tv_usec = MIN(999999, cfg->s[SRegisterGuardTime] * 20000);

From what I understand, this is because tv_usec needs to be < 1000000 or select() will error.  

The code should probably be changed to handle the error better if select() fails, but the above will fix this particular error.

**Comment by FozzTexx** [comment]
What do you mean that select is failing?

**Comment by reflectiveroot** [comment]
Looking back I found I made a couple notes regarding this issue.  From my debug output log, I saw this:

2017-03-23 18:49:08:27355648:DEBUG_X:Setting timer for break delay
2017-03-23 18:49:08:27355648:WARN:Select returned error (Invalid argument)

This error is caused by the select statement at line 367 in bridge.c.  The ptimer->tv_usec parameter was set out of range.  On some OSs the bad param is accepted or ignored, but on FreeBSD (maybe OS X?) it will return an error, and eventually the code hangs when trying to receive input from the client.

After making the change suggested in the OP, the code now works fine in FreeBSD.
